### PR TITLE
Reboot improvements, player panel fix, IsBanned fix

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Improves and consolidates the Reboot() proc. The "Immediate Reboot" proc has been removed, and replaced with the option to enter a delay time in the Restart proc: https://github.com/tgstation/-tg-station/pull/9735/https://github.com/tgstation/-tg-station/pull/11507.
2) IsBanned() now rejects blank ckeys/IP's/CID's: https://github.com/tgstation/-tg-station/pull/8177.
3) The player panel will now work again on clientless mobs.
4) Adds an option to kick all (AFK) clients from the lobby to the player panel.

### CODE REVIEW

**Comment by tigercat2000** [comment]
I don't think the immediate reboot proc should be removed, sometimes you do need to just hit a button and have the world immediately stop and restart.

File: code/world.dm [hunk 1]
```diff
@@ -215,20 +215,39 @@ var/world_topic_spam_protect_time = world.timeofday
...
+	world << "<span class='boldannounce'>Rebooting world in [delay/10] [delay > 10 ? "seconds" : "second"]. [reason]</span>"
+	sleep(delay)
+	if(blackbox)
+		blackbox.save_all_data_to_sql()
+	if(ticker.delay_end)
+		world << "<span class='boldannounce'>Reboot was cancelled by an admin.</span>"
+		return
+	feedback_set_details("[feedback_c]","[feedback_r]")
+	log_game("<span class='boldannounce'>Rebooting world. [reason]</span>")
+	kick_clients_in_lobby("<span class='boldannounce'>The round came to an end with you in the lobby.</span>", 1) //second parameter ensures only afk clients are kicked
```
Comment (tigercat2000) [inline code/world.dm]: Uh, people in the lobby shouldn't be forcefully disconnected on Reboot.

Comment (Markolie) [inline code/world.dm]: There's no need to keep them in-game, really. It stops people from joining the game, going AFK and just leaving their game on for days on end. Not that that happens often (I presume?) though.

Comment (tigercat2000) [inline code/world.dm]: We don't kick AFK clients anywhere else, and AFK does not necessarily equal inactive. There is no need to kick them, either.

Comment (Markolie) [inline code/world.dm]: You'd have to be AFK for an entire round and remain in the lobby in order to be kicked: that would mean that they are indeed inactive. There's no reason to keep them on the server.

Comment (Markolie) [inline code/world.dm]: Actually, you do have a point. /tg/ does kick AFK clients, whereas we don't. I'll get rid of it.

**Comment by tigercat2000** [comment]
As an addendum to my above comment about immediate reboot, I think a good middle would just be to leave it in and call Reboot with no delay.

**Comment by Markolie** [comment]
You can now set the delay on the Restart admin proc. If you set the delay to 0, it'll immediately restart.

**Comment by tigercat2000** [comment]
My issue with that is that the Restart admin proc pops up an input box, instead of immediately attempting to trigger a reboot.

**Comment by Markolie** [comment]
I don't think that really matters much. Insta-reboots can still be done through the "Reboot" server option by right clicking on the menu bar.

**Comment by tigercat2000** [comment]
That option isn't available to normal admins, only those with profiler access.

**Comment by Markolie** [comment]
Mhm, I'll check with Earth/Mel to make that available to all Fullmins and up.

**Comment by tigercat2000** [comment]
Might want to look at tgstation/-tg-station#11507

**Comment by Markolie** [comment]
Yup, I included that in this PR.

**Comment by MrStonedOne** [comment]
something to concider: http://www.byond.com/forum/?post=1928078

I'll be adding it to tgstation/-tg-station#11507 since it's backwards compatible with a simple if (usr)

**Comment by Markolie** [comment]
Ah, very nice, thank you! I'll keep an eye on the PR and port over the change, and then move the logging from restart() to Reboot().

**Pull request merged**